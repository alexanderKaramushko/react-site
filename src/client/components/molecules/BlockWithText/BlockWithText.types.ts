import { Themes } from '../../../common/settings';

export enum LabelSize {
    SMALL = 'small',
    MEDIUM = 'medim',
    LARGE = 'large',
}

export interface OwnProps {
    children: React.ReactNode;
    label: string;
    labelSize?: LabelSize;
    rowReverse?: boolean;
    theme?: Themes;
}

export interface StateProps {
    theme?: Themes;
}

export type DispatchProps = {}

export type Props = OwnProps & StateProps & DispatchProps;
