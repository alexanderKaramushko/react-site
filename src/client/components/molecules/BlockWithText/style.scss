.blockWithText {
    display: flex;
    align-items: center;
    padding: 5px;
}

.rowReverse {
    flex-direction: row-reverse;
}
