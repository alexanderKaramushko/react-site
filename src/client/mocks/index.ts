import configureStore, { MockStoreEnhanced } from 'redux-mock-store';

export function mockStore<T>(initialState: T): MockStoreEnhanced {
  const configureMockStore = configureStore();
  const store = configureMockStore(initialState);

  return store;
}
