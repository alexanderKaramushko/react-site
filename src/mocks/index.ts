import configureStore, { MockStoreEnhanced } from 'redux-mock-store';

export function mockStore(initialState: any): MockStoreEnhanced {
    const configureMockStore = configureStore();
    const store = configureMockStore(initialState);

    return store;
}
