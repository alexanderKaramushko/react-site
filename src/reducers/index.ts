import { combineReducers } from 'redux';
import { numberCollectionReducer } from './number-collection.reducer';
import { settingsReducer } from './settings.reducer';

export const rootReducer = combineReducers({
    numberCollectionReducer,
    settingsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
