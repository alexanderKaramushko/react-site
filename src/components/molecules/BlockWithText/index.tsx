import BlockWithText from './BlockWithText.component';

export default BlockWithText;
