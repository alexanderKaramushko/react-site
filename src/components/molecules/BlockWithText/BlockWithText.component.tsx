import React from 'react';
import classnames from 'classnames';
import { Translate } from 'react-redux-i18n';
import Label from '../../atoms/Label/Label.container';
import { Themes } from '../../../common/settings';
import Fade from '../../transitions/Fade/Fade';
import { LabelSize, Props } from './BlockWithText.types';
import * as styles from './style.scss';

const defaultProps = {
    children: '',
    label: '',
    labelSize: LabelSize.SMALL,
    rowReverse: false,
    theme: Themes.LIGHT,
};

const BlockWithText: React.FunctionComponent<Props> = (props) => {
    const {
        children, label, labelSize, rowReverse,
    } = props;
    const labelClassProps = classnames(
        styles.blockWithText,
        {
            [styles.rowReverse]: rowReverse,
        },
    );

    return (
        <Fade>
            <div className={labelClassProps}>
                <Label size={labelSize}>
                    <Translate value={label} />
                </Label>
                {children}
            </div>
        </Fade>
    );
};

BlockWithText.defaultProps = defaultProps;
BlockWithText.displayName = 'BlockWithText';

export default BlockWithText;
