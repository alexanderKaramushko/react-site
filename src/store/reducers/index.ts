import { combineReducers } from 'redux';
import { numberCollectionReducer } from './test/reducers';

export const rootReducer = combineReducers({
    numberCollectionReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
