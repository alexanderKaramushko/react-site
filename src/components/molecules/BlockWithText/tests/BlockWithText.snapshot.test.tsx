import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MockStoreEnhanced } from 'redux-mock-store';
import { mockStore } from '../../../../mocks';
import BlockWithText from '../BlockWithText.component';
import { RootState } from '../../../../reducers';

const store: MockStoreEnhanced = mockStore({
    settingsReducer: {
        themeName: 'light',
    },
} as RootState);

describe('>>>BlockWithText --- Snapshot', () => {
    it('+++capturing Snapshot of Block', () => {
        const renderedValue = renderer.create(
            <Provider store={store}>
                <BlockWithText
                    label="Test LabeledImage"
                >
                    <span>Test LabeledImage</span>
                </BlockWithText>
            </Provider>,
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
