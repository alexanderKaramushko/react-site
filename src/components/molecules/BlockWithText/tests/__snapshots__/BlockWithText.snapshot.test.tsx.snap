// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>BlockWithText --- Snapshot +++capturing Snapshot of Block 1`] = `
<div
  className=""
>
  <span
    className=""
  >
    <span>
      Test LabeledImage
    </span>
  </span>
  <span>
    Test LabeledImage
  </span>
</div>
`;
